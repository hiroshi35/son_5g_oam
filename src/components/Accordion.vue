<template>
  <div>
    <div class="panel-group" :id="id" role="tablist" aria-multiselectable="true">
      <slot></slot>
    </div>
  </div>
</template>
<script>
  export default {
    props: {
      id: {
        default: 'accordion' + Math.floor((Math.random() * 1000) + 1)
      }
    }
  }
</script>

<style lang="scss">
  .accordion {
    margin-bottom: 5px;

    .panel-heading {

      border-radius: 0;
      a {
        text-decoration: none;
      }
    }
  }
</style>
