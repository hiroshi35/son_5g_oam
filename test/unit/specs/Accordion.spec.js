import { shallow } from 'vue-test-utils'
import Accordion from '@/components/Accordion.vue'

describe('Accordion.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Accordion,
      {
        propsData: {
          id: 'accordion_id'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has correct props', () => {
    expect(wrapper.props().id).toBe('accordion_id')
  })
})
