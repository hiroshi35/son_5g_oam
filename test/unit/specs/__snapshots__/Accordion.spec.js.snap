// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion.vue has the expected html structure 1`] = `
<div>
  <div
    aria-multiselectable="true"
    class="panel-group"
    id="accordion_id"
    role="tablist"
  />
</div>
`;
